### DISCUSSION TITLE
Fighters and drones gameplay ideas - Survivabilty and retrieval

### DISCUSSION THREAD

**Discussion by kybermen** [endless-sky/endless-sky#9540] [Balance, Mechanics, and Interactions]
It would be nice if your fighters and drones have more chance to survive. Right now they are kinda iffy to use since they tend to get destroyed pretty fast, and you having to replace them a lot.

If you can make them invulnerable when disabled unless targeted, AI tend to switch targets immediately when their target is disabled, so unless your fighter got hit by a strong attack that kills them on the last hit,  this will let them survive so you can retrieve them after the battle. 

If you want enemies to kill fighters. You can also create a new "_ship personality_" that would keep targeting the fighters until they are dead. Like the _IsVindictive_ personality, but specifically for fighters. Maybe this could be the personality of some enemy fighters. 

Oh also, they should still be damaged by AoE attacks when disabled, like explosions from missiles, even those not targeting them. 
Also burn damage. 

I'm not sure how the invulnerability should affect leftover projectiles. Like when Proton gun will still kill a ship after it actually stopped shooting unlike lasers that stops instantly. But i think its better if Invulnerability starts once disabled. Or maybe random chances projectile still hit? Maybe only missiles with homing capability will still hit. Whichever it is, it would still increase the chance of them not getting destroyed.

**Comment by mOctave** [comment]
This is definitely an interesting approach to solving the fighter death issue. However, as you've touched on, there's a few issues with it:
**1. Implementation:** As you've said, there's a lot of variables that need to get decided on here. Definitely, AoE should still work, but the Proton Gun issue could be a problem. Random chance for things like this has been brought up, and has its own issues (see https://github.com/endless-sky/endless-sky/issues/7648#issuecomment-1328369948). Also, the immunity would presumably involve the attack passing over or under the fighter, not hitting it and doing nothing, right? If so, I imagine that could be a hassle to code.
**2. Realism:** This one is less of an issue, because ES isn't a very realistic game to begin with, but it still would rear its ugly head here. People would wonder why the fighters weren't dying, and likely go on to report it as a bug. It all comes back to the issue that if it looks like something should happen and it doesn't, that takes away from gameplay. This might not be much of an issue for vanilla, since fighters and drones are generally small and I could imagine targeting programs slightly missing them or something along those lines, but I can't imagine what would happen if my Pocket Warships behaved the same way because they were carried in bays. True, that's plugin content, but any solution we have shouldn't come at the _cost_ of plugins, even if it doesn't benefit them.
**3. Leftover Fighters:** Even presuming everything works perfectly (which it probably could if someone put enough effort into this), there's still the problem of the leftover fighters. If fighters and drones are immune, then there could easily be thirty or forty of them strewn about and disabled after some of the battles in the FW campaign. I can't imagine any in-game faction wanting to design their ships so that if they lose a battle they suddenly all end up in their enemies' hands. At the very least, this change could lead to a player who wasn't focused on capturing and didn't have any special H2H gear or extra crew having the full complement of a Carrier at their disposal.

With all that said, I'm not sure I agree with the idea of full immunity for all carried. However, I would be fine with a similar immunity given by an attribute, provided the end result didn't look too janky. That could allow for many very small ships to receive this, even if they weren't carried, while blocking non-vanilla and much larger content from being affected.

**Reply by kybermen** [reply]
1. Yes, that is what i meant with the immunity. I think there is one instance where it happens, I'm not sure though, doesn't some pug weapons shoot over obstacles? If it does, maybe the code for that can be modified and used?

2. Well, I'm not really sure what exactly would be defined as "Pocket Warships" but maybe only ships under certain mass will have the effect? So only really small fighters. So yeah, an attribute that can only be given to Small ships would work. 

3. For the left over fighters. Maybe use same system from Marauders or the Sestor/Mereti ships. When you go for boarding and/or capture, the pilot could decide to eject and self destruct. Or they could also roll for a self destruct when their Carrier is destroyed.

I also wouldn't like it if let say a Carried fighter the size of a Leviathan becomes invulnerable. Although that would really help a lot in capturing them. 

Also an additional idea, but also more work to implement. To support the lore on why they become invulnerable and also maybe solve the proton gun issue, upon disabling, maybe have the fighter have a small explosion that deals them no additional damage, but kicks them off on their current path and makes what ever else that was coming for them miss (proton gun). Kinda like what happens when other big ships explode, they fling all the tiny ships around to the ends of the universe.

**Reply by mOctave** [reply]
1. Yes, that could work.
2. Pocket Warships are a specific example of plugin content (like your Leviathan-size fighter, for example).
3. I don't really like the self-destruct solution. If it's the pilot's choice, then why would they want to leave their tiny vessel to go floating through empty space when they still have a chance of being rescued if they stay behind? Maybe the Navy would do this, but I doubt they'd want to scuttle their fancy toys if they didn't have to. And if it was the carrier, that could make sense for drones, but it would still seem odd for any kind of manned craft to blow up. There's a few groups in the game that I could envision doing that to their pilots, but not many.

The explosion upon disablement idea would work well gameplay-wise if it was possible to implement, but I'm not sure how it would work in terms of lore: it would look a lot like the start of the fighter being destroyed.

**Reply by kybermen** [reply]
On the Lore side, it would be pretty much how it played out in gameplay. A small explosion would make the fighter veer suddenly off course. A critical component is badly damaged that it exploded, maybe it's the engine or reactor, maybe the weapons exploded, this would normally be the time where the pilot would go "Mayday mayday, I've been hit" and start crashing to the ground, but we are in space so now the fighter's last safety measures would kick in, if its not the reactor but something else that exploded, power would be cut out form the reactor, fire extinguishers or whatever will be used to stop further damage, thus leading to "Disabled" state. The fighter suddenly veering off course might lose the enemy tracking it and since the target is "out" anyway it's better to switch to active targets, thus leading to the "immunity".

Now I'm also not a fan of the self destruct system. But the idea that a Pilot would choose to eject (maybe in an escape pod or just his space suit idk) is because it is be a Gamble. Waiting to be rescued in ES, is a gamble, especially in hostile territory, you aren't sure that the next person who will open the hatch to your ship is an enemy or not. You would be lucky if they only took parts from your ship, worse case "lore wise" you become a slave. The higher ups of each faction doesn't have to force the self destruct, and even if they do, pilots will always eject to wait for rescue, but waiting in your ship is Risky because it might attract pirates (or first last), and if you are inside, you get could either get killed or captured along with the ship. 

Imagine this, your flag ship is disabled,you might decide to wait for the battle to be over or calling one of your ship for help (Rescue) or maybe just choosing to load up the last save(Self Destruct). Should you decide to ask for help, suddenly a huge fleet of sestor and mereti jump in and starts a brawl on your position killing you, or that Palavret is slowly making it's way on your disabled ship surely trying to board it then making it go boom, or a disabled world ship next to you got hit and its now about to explode, or something along those lines, not like I've been in those situations or anything. Anyway, Self destruct is just one of the options.

Other possible option would be, making the ship uncapturable. Either like the Ka'Het, or like on Remnant missions with ships with broken jump drives not being able to leave the system. Maybe engine or power source are broken or missing, making capturing the ship pointless since you can't take it with you, since you can't Tow ships in this game. Ka'Het idea is honestly not very good, not like a password or a missing key stops anyone from taking a Quarg Wardragon home, so only missing/broken outfits seems to be viable. Honestly, there should be a ways to Capture Ka'Het, like adding the command room to drones.

**Comment by yafflehk** [comment]
I've posted this idea before but...
Rather than spend a lot of effort changing the AI, wouldn’t it be easier to hack the current system to get the kinds of behavior we want? 
**Turrets are fighters.**
Each turret is a fighter, no thrust or power, bolted to the main ship, but a fighter none the less. This sounds like a weird and pointless idea but hear me out.
This gives actual fighters a purpose for existing – if turrets are seen by the game as separate ships with their own stats and targeting, they can be disabled separately from the hull they are mounted on. Each capital ship would, in effect, become its own fleet, with each turret becoming an individual target. Fighters can now reasonably attack a capital ship (which they do now anyway without much effect) and have a decent chance to make a difference in a fight – especially if we limit some “turrets” in their choice of target. I’m thinking of the scene in Star Wars where the turrets defending the trench on the death star are destroyed. Lore would have to be that either, turrets have no shields (can’t fire through a shield, right?) or that shields are off around the turret while it fires. 
This is probably best described in a scenario:
You jump into a system, your ship is a mule with 2 heavy laser turrets and a point defense turret, let’s say it’s a modified quad blaster, it only shoots at fighters and missiles and tracks a lot faster than a standard quad blaster, let’s give it a weight penalty and crew of 2 to make up for that. You have 2 sparrows as escorts, and you have a Dagger in your single bay. On the other side of the system you see a pirate Leviathan and three wasps. The wasps immediately go to full power and streak towards you, your escorts rush to intercept them, you launch the Dagger as well. One of the wasps has heavy missiles, he one shots one of your Sparrows but the viper and the other sparrow manage to disable a wasp leaving a fairly even dogfight, which, unfortunately for you, the wasps win. The wasps are now free to attack your ship, but they are fighters, they will attack armed fighters first. They see your ship as a small fleet of an unarmed mule and three armed fighters, they begin to make passes at you as you maneuver defensively. The quad blaster disables one wasp but not before it has disabled one heavy laser and damaged the QB. The other wasp is zapped by the QB turret just before the leviathan arrives…

As you can see, without a fighter screen the Wasps would have disabled all your armament almost straight away, leaving you totally vulnerable to the Leviathan.
This system makes having a large fleet with plenty of fighter bays a reasonable gameplay decision, it means that above a certain level you MUST have fighter escorts and dedicated point defense turrets. It tricks the game into giving fighters a purpose and a useful thing to do in a battle.

**Reply by mOctave** [reply]
I'd play a game like that, happily.
However, I'm not sure it would be ES anymore.

**Reply by kybermen** [reply]
The Dagger still died... and i wouldn't want to buy replacement turrets in addition to fighters.

This would be a great idea for Super Heavies Like Korsmanath A'woj. You could try picking out the weapons first, instead of going for half a million damage while enduring a Rain of like 20 turrets and 20 fighters. Although Targeting would be a problem if turrets are place under the ship.

**Comment by QuantumTux** [comment]
@mOctave @kybermen 

I made some suggestions in Issue #9671 that are relevant to this topic.

**Comment by xX-Dillinger-Xx** [comment]
A bit late but I also like this idea. I feel fighters are actually useless after a certain point in the game. I don't buy them and don't want to capture them. Actually, disabled enemy fighter become annoying when you are trying to board disabled ships after a battle. "B" just keeps selecting all these fighters and it becomes frustrating to select what I'm really after. This idea would make fighters more appealing to some types of game play. Rather then having to replace the turrets you could repair them for a lesser cost then re purchase. Repairs could only be completed at planets with shipyards and/or outfitters if desired.

**Comment by QuantumTux** [comment]
@yafflehk Turrets are **NOT** Fighters.

Turrets don't fly away from your ship to chase after worthless targets

Turrets don't make mindless individual attacks guaranteed to get them blown up before they can do anything more than amuse the enemy

Turrets don't uselessly mill around like stoned cows

Turrets are not fighters and the solution here is not to treat them as something they are not.

**Reply by mOctave** [reply]
I believe what yafflehk was originally suggesting was that we _make_ turrets into fighters as far as the game engine sees them, not that turrets _currently_ are fighters.

**Reply by xX-Dillinger-Xx** [reply]
I agree with @mOctave. I believe he was suggesting it as a simple solution with the least amount of impact to programmers. 
From the players point of view, fighters would focus on other fighters and turrets using hit and run tactics. From the game engines point of view, if turrets were classified as fighters, it would allow fighters to focus fire on them and other fighters. However, at the moment, unless I'm mistaken ,fighters just target anything that is the enemy. Which renders them pointless. A new "ship personality" would be needed as @kybermen originally suggested.

**Comment by xX-Dillinger-Xx** [comment]
Why couldn't we just use the already implemented defensive cloaking? The Merganser already does exactly what we are talking about. It gets too damaged, cloaks, retreats to a safe distance until healed, and then re-engages. Is this not what we are talking about for fighters and drones? 
So, I see it this way, 
1. fighters and drones get deployed
2. fight
3. get damaged to a set point
4. cloak
5. head back to a carrier for repair
6. then re-deploy

EDIT: I went and created a couple of Fighters with cloaking. It doesn't quite work. The problem is they can't land in the "bays".  Instead, I get a message that says you can not board a ship while cloaked. However, they did survive longer then the non cloaking versions. They died of splash damage because they where were basically glued to me, trying to dock. I was able to uncloak them and get them to dock, but timing had to perfect on my part. I was using a Rano'erek that did not have cloaking, two smews with cloaking and two without cloaking.
